clients/privacy-center/config/config.json

```diff
@@ -43,7 +43,10 @@
         "name": "Data Sales or Sharing",
         "description": "We may use some of your personal information for behavioral advertising purposes, which may be interpreted as 'Data Sales' or 'Data Sharing' under regulations such as CCPA, CPRA, VCDPA, etc.",
         "url": "https://example.com/privacy#data-sales",
-        "default": true,
+        "default": {
+          "value": true,
+          "globalPrivacyControl": false
+        },
```

Comment:
This is the crux of the changes. [Here's the type definition](https://github.com/ethyca/fides/pull/2341/files#diff-103fdb8585246217bff5eb7f5eec4f86a0c93947dc122941b7952615c17dfce2R20) for the new config option. Note that a plain boolean is also valid and would be handled as if GPC doesn't matter, as before.

An alternative implementation would be to keep `default: boolean` and add a sibling config like `defaultWhenGlobalPrivacyControlEnabled: boolean` or something. That version seems more self-contradictory: "default" wouldn't really be the default 😬. With the `ConsentValue` object, we'll be able to add more options if/when we need them.

Reply:
Does this configuration mean the default value for this consent option is true (Opted-in) and GPC signal doesn't apply to this consent option?

If GPC applies to this consent option, should the engineer configure this option as true?

Reply:
https://github.com/ethyca/fides/pull/2341/files/05e70196abb4bcd153033d412646820d9a4d1922#r1084674181

> Does this configuration mean the default value for this consent option is true (Opted-in) and GPC signal doesn't apply to this consent option?

If GPC applies to this consent option, should the engineer configure this option as true?

There are two ways we could make GPC configurable:

**1: "Is this affected by GPC?"**

This sounds like what you were expecting the true/false to indicate. I think this is the intuitive question when looking for a quick-fix for respecting GPC, but it has some downsides:

1. It's hard to come up with a name that makes it clear how this config would work: `respectsGlobalPrivacyControl` or `affectedByGPC` ... Let me know if you have any ideas. Let's call it `affectedByGPC` for now. 
2. It is confusing setting `affectedByGPC: true` would mean _setting the default consent to `false`_. It is very easy to introduce confusing double-negatives in these features.
3. It seems likely that we will have _other_ settings that affect the default consent value. For example, location-based checks for EU, California, or Virginia. If we put `affectedByGPC` on the [ConfigConsentOption](https://github.com/ethyca/fides/blob/b4cadae4cb15794718508857ca8ee0bbbae5ec4f/clients/privacy-center/types/config.ts#L28), then adding another config that can mess with the `default` config is even more confusing. That's why my current implementation puts these conditions under the `default` config itself: this makes it clear (I hope) that the conditions only affect how the default consent is set.

**2: "If GPC is active, what should the default be?"**

This is how we should think of the GPC config question (IMO). This helps us address each of the downsides listed above:

1. The config called `globalPrivacyControl` refers to the situation when GPC is active. Maybe this could still benefit from a more detailed name.
2. Reading the config file now makes it clear that the _default_ value changes based on GPC:
  ```json
  "default": {
    "value": true,
    "globalPrivacyControl": false
  }
  ```
3. This also opens up other ways to changes the default consent based on other data-subject/browser information (what I'm calling "consent context"). For example:
  ```json
  "default": {
    "value": true,
    "globalPrivacyControl": false,
    "location": {
      "europe": false,
      "virginia": false,
      "california": true,
    }
  }
  ```

That's just hypothetical, but I think it makes clearer how the default consent is chosen based on the user's browser.

Reply:
Thanks @ssangervasi for the explanation. It's clear for me and I agree this is the best way to make it extensible for the future and manage code. 
Separately, we'll have to make this super clear in the documentation.